src/Microsoft.Extensions.Logging.Abstractions/LoggerMessage.cs

```diff
@@ -0,0 +1,343 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.Extensions.Logging.Internal;
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Extensions.Logging
+{
+    /// <summary>
+    /// Creates delegates which can be later cached to log messages in a performant way.
+    /// </summary>
+    public static class LoggerMessage
```

Comment:
Yeah not a huge fan of the name either. Usually these types of static classes are more verby and less nouny because they merely contain helper methods, and do not themselves represent anything.